src/e2e/java/teammates/e2e/cases/e2e/AdminSessionsPageE2ETest.java

```diff
@@ -0,0 +1,181 @@
+    private Instant instantTomorrow = TimeHelper.getInstantDaysOffsetFromNow(1);
+    private Instant instant3DaysLater = TimeHelper.getInstantDaysOffsetFromNow(3);
+    private Instant instantNextWeek = TimeHelper.getInstantDaysOffsetFromNow(7);
+    private Instant instant10DaysLater = TimeHelper.getInstantDaysOffsetFromNow(10);
+    private Instant instant14DaysLater = TimeHelper.getInstantDaysOffsetFromNow(14);
+    private Instant instant24DaysLater = TimeHelper.getInstantDaysOffsetFromNow(24);
+
+    private String formatDateTime(Instant instant, String timeZone) {
+        return DateTimeFormatter
+                .ofPattern("EEE, dd MMM YYYY, hh:mm a")
+                .format(instant.atZone(ZoneId.of(timeZone)))
+                .replaceFirst(" AM$", " am")
+                .replaceFirst(" PM$", " pm");
+    }
+
+    @Override
+    protected void prepareTestData() {
+        testData = loadDataBundle("/AdminSessionsPageE2ETest.json");
+
+        // To guarantee that there will always be some "ongoing sessions" listed,
+        // the test data is injected with date/time values relative to the time where the test takes place
+
+        openFeedbackSession = testData.feedbackSessions.get("session1InCourse1");
+        openFeedbackSession.setStartTime(instant3DaysAgo);
+        openFeedbackSession.setCreatedTime(instant3DaysAgo);
+        openFeedbackSession.setSessionVisibleFromTime(instant3DaysAgo);
+        openFeedbackSession.setEndTime(instant3DaysLater);
+        openFeedbackSession.setResultsVisibleFromTime(instant3DaysLater);
+
+        awaitingFeedbackSession = testData.feedbackSessions.get("session2InCourse1");
+        awaitingFeedbackSession.setStartTime(instantTomorrow);
+        awaitingFeedbackSession.setCreatedTime(instant3DaysAgo);
+        awaitingFeedbackSession.setSessionVisibleFromTime(instantTomorrow);
+        awaitingFeedbackSession.setEndTime(instant3DaysLater);
+        awaitingFeedbackSession.setResultsVisibleFromTime(instant3DaysLater);
+
+        futureFeedbackSession = testData.feedbackSessions.get("session3InCourse1");
+        futureFeedbackSession.setStartTime(instant10DaysLater);
+        futureFeedbackSession.setCreatedTime(instant3DaysAgo);
+        futureFeedbackSession.setSessionVisibleFromTime(instant10DaysLater);
+        futureFeedbackSession.setEndTime(instant24DaysLater);
+        futureFeedbackSession.setResultsVisibleFromTime(instant24DaysLater);
+
+        removeAndRestoreDataBundle(testData);
+    }
+
+    @Test
+    public void testAll() {
+
+        ______TS("verify loaded data");
+
+        AppUrl sessionsUrl = createUrl(Const.WebPageURIs.ADMIN_SESSIONS_PAGE);
+        AdminSessionsPage sessionsPage = loginAdminToPage(sessionsUrl, AdminSessionsPage.class);
+        sessionsPage.waitUntilAnimationFinish();
+
+        String tableTimezone = sessionsPage.getSessionsTableTimezone();
+
+        List<WebElement> ongoingSessionRows = sessionsPage.getOngoingSessionsRows();
```

Comment:
Is it possible to shift this verification into `AdminSessionsPage` as I believe all interaction with WebElements should be hidden in the page object and we should avoid having getters for any WebElements. It can also be reused for the below testcase.